### PR TITLE
Stackrox ibmcloudz

### PR DESCRIPTION
**Pull request by gbhure**
Configurations to create/destroy k8s cluster in IBM Cloud for Z. The required secret ibmcloudz-rhr-creds is already added to the vault with IBM Cloud details.

The goal here is for IBM z platform enablement for stackrox ci, i.e., to create and destroy IBM z K8s cluster on IBM Cloud as the first step. IBM Cloud currently doesn't support k8s z cluster management by default. So we have to remotely log into an intermediate node (163.66.94.115), and perform a installation/destroy through script called k8smanager, which is already set up there. This is a very similar process to the IBM power platform enablement: [498265a](https://github.com/openshift/release/commit/498265a7f38b7bdc0420a9aec0331f438555cfb9)

/cc @gavin-stackrox

### CODE REVIEW

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml [hunk 1]
```diff
@@ -696,6 +696,64 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )gke-version-compatibility-tests,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
[the remark this answers is not in the document]
Comment (gbhure) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-presubmits.yaml]: Done
/cc @gavin-stackrox

File: ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ref:
+  as: stackrox-automation-flavors-ibmcloudz-e2e-cluster-create
+  commands: stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-commands.sh
+  credentials:
+  - mount_path: /tmp/vault/ibmcloudz-rhr-creds
+    name: ibmcloudz-rhr-creds
```
Comment (gavin-stackrox) [inline ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-ref.yaml:6]: I think this name needs to be the name configured in `secretsync/target-name`

**Comment by gavin-stackrox** [comment]
@gbhure I updated the secret. Can you try a rebase and `make update` to get the `ci/prow/ordered-prow-config` job to pass? I will approve the PR once that is ready and we can work on the job with stackrox/stackrox PRs.
Thanks

**Pull request merged**